### ISSUE TITLE
Reference Input has required attribute - resource, but in docs there are not information about it

### ISSUE THREAD

**Issue by pullrequestdemoffice** [marmelab/react-admin#2786]
if i pass object {
    component: ReferenceInput,
    label: 'Price List',
    source: 'priceListId',
    children: <SelectInput optionText="description" />,
    resource: 'priceList',
    reference: 'priceList',
  }

**Comment by pullrequestdemoffice** [comment]
if i pass object {
component: ReferenceInput,
label: 'Price List',
source: 'priceListId',
children: ,
resource: 'priceList',
reference: 'priceList',
} to render ReferenceInput i have error with required resource props, i also have a resource with this reference in my app.js ( <Resource name="priceList" /> )

**Comment by yuriydobryanskyyempeek** [comment]
let me rephrase: looks like `ReferenceInput` has a required `resource` prop while documentation doesn't say anything about it. It says about `source` and `reference` being mandatory, but does not say anything about `resource`. What is the `resource` prop for? If it is indeed a required prop - than I think we should make it described in the documentation

**Comment by yuriydobryanskyyempeek** [comment]
Yep, that was our problem. Apparently when we created custom form layout we did not pass all the props down to input components. I think you can close the issue
